mesonbuild/compilers/compilers.py

```diff
@@ -296,6 +296,14 @@ class CompileCheckMode(enum.Enum):
     OptionKey('b_vscrt'): coredata.UserComboOption('VS run-time library type to use.',
                                                    ['none', 'md', 'mdd', 'mt', 'mtd', 'from_buildtype', 'static_from_buildtype'],
                                                    'from_buildtype'),
+    OptionKey('b_vstoolset'): coredata.UserComboOption('VS toolset to use.',
+                                                   ['none', 'v90', 'v100', 'v110', 'v110_xp', 'v120', 'v120_xp',
+                                                   'v140', 'v140_xp', 'v140_clang_c2', 'LLVM-vs2012', 'llvm-VS2012_xp',
+                                                   'LLVM-vs2013', 'LLVM-vs2013_xp', 'LLVM-vs2014', 'LLVM-vs2014_xp',
+                                                   'LLVM-vs2017', 'LLVM-vs2017_xp', 'v141', 'v141_xp', 'v141_clang_c2',
+                                                   'LLVM', 'LLVM_v141', 'LLVM_v141_xp', 'LLVM_v142',
+                                                   'v142', 'v143'],
+                                                   'none'),
```

Comment:
The naming scheme here has inconsistent capitalization, underscore vs hyphen, etc. At the very least it should be all lowercase and use either hyphens or underscores. Not both interchangeably.

Reply:
This is the name of vs own toolset. I didn’t name it. I try to collect it comprehensively.

Reply:
I have removed clang now, so that the naming scheme is consistent, and most of what we want to use is the remaining vstoolset.